### PR TITLE
Add icon only docs for Button

### PR DESCRIPTION
**Pull request by jluyau**
Closes 

Add icon-only entry for Button in docs 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/button/docs/Button.mdx
Comment (devongovett) [inline packages/@react-spectrum/button/docs/Button.mdx:171]: The buttons need `aria-label` props.

**Comment by devongovett** [comment]
Noticed that the padding is somehow off on the Events example now: https://reactspectrum.blob.core.windows.net/reactspectrum/eab25a55e7e4b7b5895d0670e92202c95d6b5057/docs/react-spectrum/Button.html#events

File: packages/@react-spectrum/button/src/Button.tsx
Comment (jluyau) [inline packages/@react-spectrum/button/src/Button.tsx:54]: Add this conditional back in to make the icon-only scenario more specific because of the [Button Events example](https://react-spectrum.adobe.com/react-spectrum/Button.html#events).

In that case, React is splitting the children into a array -- [number, string], so the Button code will render that since it doesn't qualify `typeof children === 'string'`. Since it renders as is, the `.spectrum-Button-label` doesn't get applied, and with the previous code, it would apply the icon only styling, thus removing the min-width/padding.

Comment (devongovett) [reply packages/@react-spectrum/button/src/Button.tsx:54]: Seems fine. Unfortunately, you'll have to move this code change to a separate pr so we can merge it prior to release. The docs have to wait until after release.

**Review by ktabors — APPROVED** [review]
Confirming that some buttons look weird, but will be fixed once #3503 is merged.

Example of weird is the "Events" example button in the docs
image

**Review by ktabors — COMMENTED** [review]
Do we need to be explicate about "If the label is hidden, an icon is required, and the label will appear in a tooltip." or add an example of this in the Tooltip (or another section) docs?

**Comment by LFDanLu** [comment]
+1 to what @ktabors mentioned. I think a specific callout to the "If the label is hidden, an icon is required, and the label will appear in a tooltip" portion of the Spectrum guidelines is appropriate here in case people don't click on that guideline link. It would also be good to have the examples follow this practice as well.

**Comment by devongovett** [comment]
> and the label will appear in a tooltip

this is not true of our implementation however.

**Comment by LFDanLu** [comment]
right, we currently rely on the user to provide the tooltip... Is the tooltip mandatory for icon only buttons? If so, we could wrap it in a TooltipTrigger for the user but we would need them to provide a tooltip text value through a different prop. If it isn't mandatory but strongly recommended IMO we should have the example display that. Otherwise we can keep it as is

File: packages/@react-spectrum/button/docs/Button.mdx
Comment (majornista) [reply packages/@react-spectrum/button/docs/Button.mdx]: We should also provide examples of buttons with both an icon and a label, where the icon has meaning that is separate from the label.

Comment (majornista) [reply packages/@react-spectrum/button/docs/Button.mdx]: For example:

```html
<Button variant="primary">
  <Add  aria-label="Add" />
  <Text>Event</Text>
</Button>
```

or

```html
<Button variant="primary">
  <Text>Add</Text>
  <Comment  aria-label="Comment" />
</Button>

<Button variant="primary">
  <Text>Remove</Text>
  <Comment  aria-label="Comment" />
</Button>
```

**Comment by dannify** [comment]
I wouldn't add Tooltip to Button. We could add an example though.

**Review by LFDanLu — APPROVED** [review]

**Pull request merged**
